src/framework/firebase-auth/strategies/password/firebase-password.strategy.ts

```diff
@@ -0,0 +1,137 @@
+ */
+
+import { Injectable } from '@angular/core';
+
+import { Observable, of as observableOf } from 'rxjs';
+import {
+  firebasePasswordStrategyOptions,
+  NbFirebasePasswordStrategyOptions,
+} from './firebase-password-strategy.options';
+import { fromPromise } from 'rxjs/internal-compatibility';
+import { catchError, map, switchMap, take } from 'rxjs/operators';
+import { User } from 'firebase';
+import { NbAuthStrategyClass } from '../../../auth/auth.options';
+import { NbAuthStrategyOptions } from '../../../auth/strategies/auth-strategy-options';
+import { NbAuthResult } from '../../../auth/services/auth-result';
+import { NbFirebaseBaseStrategy } from '../base/firebase-base.strategy';
+
+
+@Injectable()
+export class NbFirebasePasswordStrategy extends NbFirebaseBaseStrategy {
+
+  protected defaultOptions: NbFirebasePasswordStrategyOptions = firebasePasswordStrategyOptions;
+
+  static setup(options: NbFirebasePasswordStrategyOptions): [NbAuthStrategyClass, NbAuthStrategyOptions] {
+    return [NbFirebasePasswordStrategy, options];
+  }
+
+  authenticate({ email, password }: any): Observable<NbAuthResult> {
+    const module = 'login';
+    return fromPromise(this.afAuth.signInWithEmailAndPassword(email, password))
+      .pipe(
+        switchMap((res) => this.processSuccess(res, module)),
+        catchError((error) => this.proccessFailure(error, module)),
+      );
+  }
+
+  refreshToken(data?: any): Observable<NbAuthResult> {
+    const module = 'refreshToken';
+    return this.afAuth.authState
+      .pipe(
+        take(1),
+        switchMap(user => {
+          if (user == null) {
+            return observableOf(new NbAuthResult(
+              false,
+              null,
+              this.getOption(`${module}.redirect.failure`),
+              ['There is no logged in user so refresh of id token isn\'t possible'],
+            ));
+          }
+          return this.refreshIdToken(user, module);
+        }),
+      );
+  }
+
+  register({ email, password }: any): Observable<NbAuthResult> {
+    const module = 'register';
+    return fromPromise(this.afAuth.createUserWithEmailAndPassword(email, password))
+      .pipe(
+        switchMap((res) => this.processSuccess(res, module)),
+        catchError((error) => this.proccessFailure(error, module)),
+      );
+  }
+
+  requestPassword({ email }: any): Observable<NbAuthResult> {
+    const module = 'requestPassword';
+    return fromPromise(this.afAuth.sendPasswordResetEmail(email))
+      .pipe(
+        map(() => {
+          return new NbAuthResult(
+            true,
+            null,
+            this.getOption(`${module}.redirect.success`),
+            [],
+            this.getOption(`${module}.defaultMessages`),
+          );
+        }),
+        catchError((error) => this.proccessFailure(error, module)),
+      );
+  }
+
+  resetPassword({ password }): Observable<NbAuthResult> {
```

Comment:
in the original password strategy `reset` means `update after request`, UI that a user sees when accessing "reset password link" in reset email
so there is no way they can be logged in at that moment